### PR TITLE
Ad7124 4sdz

### PR DESCRIPTION
**Pull request by cristiantpop**
Add demo project

### CODE REVIEW

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#include "error.h"
+
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+int main(void)
+{
+    struct ad7124_dev *dev; /* A driver handle to pass around */
```
Comment (buha) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:22]: please use tabs for indentation

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		/* Read data from the ADC */
+		ret = ad7124_wait_for_conv_ready(dev, timeout);
+		if (ret < 0)
+			return ret;
+
+		ret = ad7124_read_data(dev, &sample);
+		if (ret < 0)
+			return ret;
+
+		printf("ADC%"PRIi32" \n", sample);
```
Comment (buha) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:69]: is this the only thing that this project should do ? afaik we want IIO support in all projects in the `projects` folder, this PR looks like WIP

Comment (cristiantpop) [reply projects/ad7124-4sdz/src/ad7124-4sdz.c:69]: This pull request was intended for someone who asked a question on Engineering Zone, it is sopose to provide an example on how to build ad7124 driver.

**Comment by cristiantpop** [comment]
V2:

- Fix astyle

**Comment by cristiantpop** [comment]
V3:

- Remove some unused headers

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+int main(void)
+{
+	struct ad7124_dev *dev;		/* A driver handle to pass around */
+	enum ad7124_registers regNr;	/* Variable to iterate through registers */
+	long timeout = 1000;		/* Number of tries before a function times out */
```
Comment (amiclaus) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:15]: Let's use `uint32_t` / `int32_t` etc.

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+		ret = ad7124_read_data(dev, &sample);
+		if (ret < 0)
+			return ret;
+
+		printf("ADC%"PRIi32" \n", sample);
+	}
+	printf("Success\n\r");
+
+	return 0;
```
Comment (amiclaus) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:63]: `#include "error.h"` -> `return SUCCESS;`

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+	struct ad7124_init_param ad7124_init_param = {
+		.spi_init = &spi_init,
+		.regs = ad7124_regs,
+		.spi_rdy_poll_cnt = 25000,
+	};
+
+	/* Initialize AD7124 device. */
+	ret = ad7124_setup(&dev, &ad7124_init_param);
+	if (ret < 0)
```
Comment (amiclaus) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:39]: if the return value for the function is either 0 or a negative value, we can use `ret != SUCCESS`

Comment (amiclaus) [reply projects/ad7124-4sdz/src/ad7124-4sdz.c:39]: Check through the entire code.

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 6]
```diff
@@ -0,0 +1,64 @@
+#include <stdio.h>
+#include <inttypes.h>
+#include "ad7124.h"
+#include "ad7124_regs.h"
+#include "spi_extra.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
```
Comment (amiclaus) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:8]: no need, since there are no definitions.

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+		.spi_rdy_poll_cnt = 25000,
+	};
+
+	/* Initialize AD7124 device. */
+	ret = ad7124_setup(&dev, &ad7124_init_param);
+	if (ret < 0)
+		return ret;
+
+	/* Read all registers */
+	for (regNr = AD7124_Status; (regNr < AD7124_REG_NO) && !(ret < 0); regNr++) {
```
Comment (amiclaus) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:43]: I believe using camel case is outside of the convention.

`regNr` -> `reg_nr`

**Review by amiclaus — CHANGES_REQUESTED** [review]

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 8]
```diff
@@ -0,0 +1,64 @@
+#include <stdio.h>
```
Comment (mchindri) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:1]: nitpick: Shouldn't copyright be added?

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+#include "ad7124_regs.h"
+#include "spi_extra.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+int main(void)
+{
+	struct ad7124_dev *dev;		/* A driver handle to pass around */
```
Comment (mchindri) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:10]: nitpick: can you indent variable names. This way is easier to differentiate name from type

**Comment by cristiantpop** [comment]
V4:

- use uint32_t / int32_t etc

- #include "error.h" -> return SUCCESS;

- ret != SUCCESS

- regNr -> reg_nr

File: projects/ad7124-4sdz/src/ad7124-4sdz.c [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+		/* Read data from the ADC */
+		ret = ad7124_wait_for_conv_ready(dev, timeout);
+		if (ret != SUCCESS)
+			return ret;
+
+		ret = ad7124_read_data(dev, &sample);
+		if (ret != SUCCESS)
+			return ret;
+
+		printf("ADC%"PRIi32" \n", sample);
```
Comment (mchindri) [inline projects/ad7124-4sdz/src/ad7124-4sdz.c:56]: Nitpick:
```suggestion
		printf("ADC : %"PRIi32" \n", sample);
```
Or something to separet ADC from the value
or something to separate ADC from value

**Review by mchindri — APPROVED** [review]

**Comment by cristiantpop** [comment]
V5:

- Add copyright
- Indent variable names
- Nicer printf

**Review by amiclaus — APPROVED** [review]

**Pull request merged**